src/views/instances/list/instances-list.tsx

```diff
@@ -12,28 +12,37 @@ import {
     IconAdd24,
     NoticeBox,
     IconWorld24,
+    Checkbox
 } from '@dhis2/ui'
 import type { FC } from 'react'
 import Moment from 'react-moment'
 import { useNavigate } from 'react-router-dom'
-import { useAuthAxios } from '../../../hooks'
-import { GroupsWithDeployments } from '../../../types'
 import styles from './instances-list.module.css'
 import InstanceTag from './instance-tag'
 import { OpenButton } from './open-button'
 import { Heading, MomentExpiresFromNow } from '../../../components'
 import { DeleteButton } from './delete-menu-button'
+import useDeployments from './useDeployments'
 export const InstancesList: FC = () => {
     const navigate = useNavigate()
-    const [{ data, error, loading }, refetch] = useAuthAxios<GroupsWithDeployments[]>('/deployments', {
-        useCache: false,
-    })
+    const { data, error, loading, refetch, showOnlyMyInstances, setShowOnlyMyInstances } = useDeployments()
+
+    const handleCheckboxChange = () => {
+        setShowOnlyMyInstances(!showOnlyMyInstances)
+    }
     return (
         <div className={styles.wrapper}>
             <Heading title="All instances">
                 <Button icon={<IconAdd24 />} onClick={() => navigate('/instances/new')}>
                     New instance
                 </Button>
+                <div>
+                    <Checkbox
```

Comment:
Should this be a component of its own?

Reply:
It could be, but since we moved most of the logic to the useDeployment file, I didn’t think it necessary,.

Reply:
I have added some checks as well in the new commits